### PR TITLE
python: make 3.12 the latest python-3

### PR DESCRIPTION
**Pull request by imjasonh**
A future change will rebuild pip/setuptools/installer with this latest 3.12 and make those the latest for py3-pip et al.

### CODE REVIEW

File: py3.12-installer.yaml [hunk 1]
```diff
@@ -6,9 +6,8 @@ package:
   copyright:
     - license: "MIT"
   dependencies:
-    # TODO(jason): Uncomment when py-3.12 is default everywhere.
-    #provides:
-    #  - py3-installer=${{package.version}}
+    provides:
+      - py3-installer=${{package.version}}
```
Comment (mattmoor) [inline py3.12-installer.yaml:10]: these all need epoch bumps

File: meson.yaml [hunk 2]
```diff
@@ -7,17 +7,18 @@ package:
     - license: Apache-2.0
   dependencies:
     runtime:
-      - python3
+      - python-3.11
```
Comment (mattmoor) [inline meson.yaml:10]: Should this bump epoch?

File: meson.yaml [hunk 3]
```diff
@@ -7,17 +7,18 @@ package:
...
 
 environment:
   contents:
     packages:
       - busybox
       - ca-certificates-bundle
-      - py3-setuptools
-      - python3
+      - py3.11-setuptools
       - samurai
```
Comment (mattmoor) [inline meson.yaml:19]: Is this not the standard way we bootstrap?

Comment (kaniini) [reply meson.yaml:19]: I think this is perhaps a little suboptimal and we might want to look at making the solver in wolfictl a little more intelligent to deal with this edge case, but seems fine for now.

**Review by kaniini — APPROVED** [review]

**Pull request merged**